packages/ilib-loctool-regex/test/RegexFile.test.js

```diff
@@ -0,0 +1,866 @@
+                    },
+                    {
+                        // regular string that contains escaped Unicode characters. We include the "u" flag
+                        // in the regular expression flags
+                        "expression": "\\$u\\s*\\(\\s*[\'\"](?<source>[^\'\"]*)[\'\"]\\s*\\)",
+                        "flags": "gu",
+                        "datatype": "javascript",
+                        "resourceType": "string",
+                        "keyStrategy": "hash"
+                    }
+                ]
+            },
+            "**/*.tmpl": {
+                "resourceFileType": "ilib-loctool-javascript-resource",
+                "template": "resources/Translation[locale].json",
+                "sourceLocale": "en-US",
+                "expressions": [
+                    {
+                        // example:
+                        // {* @L10N This comment is on the same line *} {'Your password change is cancelled.'|f:'login_password_change_cancelled'}
+                        "expression": "\\{\\*.*@L10N\\s*(?<comment>[^*]*)\\*\\}.*\\{.*'(?<source>[^']*)'\\s*\\|\\s*f:\\s*'(?<key>[^']*)'.*\\}",
+                        "flags": "g",
+                        "datatype": "template",
+                        "resourceType": "string"
+                    },
+                    {
+                        // example:
+                        // {* @L10N The message shown to users whose passwords have just been changed *}
+                        // {'Your password was changed. Please log in again.'|f:'login_success_password_changed'}
+                        "expression": "\\{\\*.*@L10N\\s*(?<comment>[^*]*)\\*\\}.*\\n.*\\{.*'(?<source>[^']*)'\\s*\\|\\s*f:\\s*'(?<key>[^']*)'.*\\}",
+                        "flags": "g",
+                        "datatype": "template",
+                        "resourceType": "string"
+                    },
+                    {
+                        // example:
+                        // {'Your password was changed. Please log in again.'|f:'login_success_password_changed'}
+                        "expression": "\\{.*'(?<source>[^']*)'\\s*\\|\\s*f:\\s*'(?<key>[^']*)'.*\\}",
+                        "flags": "g",
+                        "datatype": "template",
+                        "resourceType": "string"
+                    }
+                ]
+            }
+        }
+    },
+    "php": {
+        "localeMap": {
+            "en-US": "EnUS",
+            "en-GB": "EnGB",
+            "de-DE": "DeDE",
+            "fr-FR": "FrFR",
+            "ja-JP": "JaJP"
+        },
+        "sourceLocale": "en-US"
+    }
+});
+
+var p2 = new CustomProject({
+    id: "emptyapp",
+    plugins: [require.resolve("../.")],
+    sourceLocale: "en-US"
+}, "./test/testfiles", {
+    "locales":["en-GB", "de-DE", "fr-FR", "ja-JP"],
+    "regex": {
+        "mappings": {
+            "**/*.js": {
+                "template": "resources/strings_[locale].json"
```

Comment:
Even if it doesn't relate to the testcase running, shouldn't the `resourceFileType`  be present for the js file type? for providing the correct example

Reply:
Well, if you do that, then when you run the unit tests, it will try to load that plugin. That means it needs to have that plugin available. I don't want the plugins to be dependent on each other unless they have to.

The regex plugin is intended for string extraction only -- you can parse all sorts of different file types with it, so you don't know how to write them out again. So for unit testing, we really only need to test the extraction part.